### PR TITLE
improve ux for robot importer check assets

### LINKED ISSUE #617: Improve Robot Importer UX
**Is your feature request related to a problem? Please describe.**
During robot importing before the CheckAssetsPage there is a wait time after clicking the next button. After clicking in the background all mesh files are being copied and preprocessed. The robot importer windows seem to be frozen during that time and there is no indication that something is happening in the background.

**Describe the solution you'd like**
I propose a change to the UX. After clicking next there is already a nice table with some information from the asset processor about the imported assets. I suggest adding additional information that would inform the user if the file is: waiting for processing, coping, processing, and done.

### PR DESCRIPTION
**Pull request by arturkamieniecki**
This PR depends on #616.
Resolves #617.

This PR improves the UX of the robot importer, by creating a separate thread for copying files. 
Usage example:

https://github.com/o3de/o3de-extras/assets/130671280/a53e20f8-ea45-43ef-b721-226621960d84

### CODE REVIEW

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h [hunk 1]
```diff
@@ -43,32 +46,52 @@ namespace ROS2::Utils
...
+        Waiting, //! Waiting for copy
+        Copying, //! Copying
+        Copied, //! Copied
+        Exists, //! Already exists
+        Failed, //! Failed
+    };
+
+    //! Bitfield containing the types of asset references are associated with a given unresolved URI or path reference.
+    //! These are flags because the same mesh URI can refer to both a Visual and a Collider entry, for example.
+    enum class ReferencedAssetType
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h:62]: This code is already in code base, why it is marked as part of this contribution.

**Review by michalpelka — COMMENTED** [review]
I like the idea, please wait for merge of 616.

File: Gems/ROS2/Code/Source/RobotImporter/Pages/CheckAssetPage.cpp [hunk 2]
```diff
@@ -94,77 +97,49 @@ namespace ROS2
...
-        if (isOk)
-        {
-            m_assetsUuidsToColumnIndex[assetUuid] = rowId;
-        }
+        m_assetsToColumnIndex[unresolvedFileName] = rowId;
     }
 
-    void CheckAssetPage::StartWatchAsset()
+    void CheckAssetPage::StartWatchAsset(
+        AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetMap, AZStd::shared_ptr<AZStd::mutex> urdfAssetMapMutex)
```
[the remark this answers is not in the document]
Comment (arturkamieniecki) [reply Gems/ROS2/Code/Source/RobotImporter/Pages/CheckAssetPage.cpp:137]: I've concluded that creating such map would not be sufficient as the mutex not only protects gets and inserts to the map but also direct changes to the elements by references. Creating such a map would require to create a custom methods for updating the elements with thread protection.

**Comment by arturkamieniecki** [comment]
I've modified the code to not depend on mutexes. The asset page is now updated directly.

**Comment by adamdbrw** [comment]
@arturkamieniecki could you rebase on new changes?

**Review by michalpelka — APPROVED** [review]
I like new approach.

**Comment by arturkamieniecki** [comment]
Rebased on top of new changes.

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h [hunk 3]
```diff
@@ -14,6 +14,7 @@
 #include <AzCore/Asset/AssetManagerBus.h>
 #include <AzCore/IO/FileIO.h>
 #include <AzCore/IO/Path/Path.h>
+#include <AzCore/IO/Path/Path_fwd.h>
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h:17]: This looks to be an unnecessary include. Please remove & test.

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h [hunk 4]
```diff
@@ -158,6 +189,47 @@ namespace ROS2::Utils
...
+    AZ::Outcome<ImportedAssetsDest> PrepareImportedAssetsDest(
+        const AZStd::string& urdfFilename,
+        AZStd::string_view outputDirSuffix = "",
+        AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetInstance());
+
+    //! Remove temporary directory for imported assets.
+    //! @param tmpDir - path to temporary directory
+    //! @param fileIO - instance to fileIO class
+    //! @returns true if succeed
+    AZ::Outcome<bool> Remove$tmpDir(const AZ::IO::Path $tmpDir, AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetInstance());
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h:231]: Please change to RemoveTmpDir

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**